includes/base/controls-stack.php

```diff
@@ -431,9 +444,41 @@ public function add_control( $id, array $args, $options = [] ) {
 			}
 		}
 
+		if ( $should_optimize_controls ) {
+			unset(
+				$args['label_block'],
+				$args['label'],
+				$args['tab'],
+				$args['options'],
+				$args['separator'],
+				$args['size_units'],
+				$args['range'],
+				$args['render_type'],
+				$args['toggle'],
+				$args['ai'],
+				$args['label_on'],
+				$args['label_off'],
+				$args['labels'],
+				$args['handles']
+			);
+		}
```

Comment:
might be easier (and safer?) to have a whitelist here of the data you actually need?
with the current approach, you'll have to maintain this list each time someone (and maybe 3rd party) adds a property

Reply:
@KingYes Again it's a blacklist vs whitelist 😉 hehe

Reply:
@StyleShit Why did you think it's not safe? As Elementor controls, I should know which controls we use on the frontend and which on the editor. No?

Reply:
I mean "safe" in terms of future proof, maintainability, and performance - If you have a whitelist of controls, you know that those controls are the only ones that will be loaded to the memory, regardless of whatever we and/or 3rd parties have added. 

This way, you can know for sure that in any future cases, only the controls that you actually need are there.

Let's say we had this PR merged before the AI was introduced, you would have needed to come back here and add the "ai" key to the blacklist or you'll gradually hurt the performance without notice (and we both know that we'll forget to come back here).

I think it's also easier to understand and determine which controls are required and affect the rendering when reading the code as a whitelist.

Reply:
@StyleShit It's a good point! But it might be tricky to construct a full whitelist without missing **anything** 🤔